scipy/optimize/_differentialevolution.py

```diff
@@ -310,6 +370,55 @@ class DifferentialEvolutionSolver(object):
         ``np.std(pop) <= atol + tol * np.abs(np.mean(population_energies))``,
         where and `atol` and `tol` are the absolute and relative tolerance
         respectively.
+
+    Attributes
+    ----------
+    x: ndarray
+        The best solution from the solver
+    convergence: ndarray
+        The standard deviation of the population energies divided by their
+        mean.
+    population: ndarray
+        The population of candidate solutions
+    result: OptimizeResult
+        The result of the optimization process (set upon context manager exit).
+
+    Methods
+    -------
+    converged()
+        Return True if the solver has converged within specified tolerance.
+
+    Examples
+    --------
+    Let us consider the problem of minimizing the Rosenbrock function. This
+    function is implemented in `rosen` in `scipy.optimize`.
+
+    >>> from scipy.optimize import rosen, DifferentialEvolution
+    >>> bounds = [(0,2), (0, 2), (0, 2), (0, 2), (0, 2)]
+    >>> with DifferentialEvolution(rosen, bounds) as solver:
```

Comment:
Would be nice if this example showed some customization of the default behavior. Because this example would be better done with ``differential_evolution``.